addons/web_editor/controllers/main.py

```diff
@@ -537,7 +537,17 @@ def modify_image(self, attachment, res_model=None, res_id=None, name=None, data=
             fields['res_id'] = res_id
         if name:
             fields['name'] = name
-        attachment = attachment.copy(fields)
+
+        # Restricted editors can handle attachments related to records to which they have access
+        # Would user be able to read fields of original record ?
+        if attachment and attachment.res_model and attachment.res_id:
+            request.env[attachment.res_model].browse(attachment.res_id).with_user(request.env.user).check_access_rights('read')
```

Comment:
Isn't this already ensured by the route model converter?
When `/web_editor/modify_image/<model("ir.attachment"):attachment>` is called, the dispatcher is getting the record to transform the ID provided by the rpc / browser into a python orm record, which should fail / raise a 403 if you don't have access.

Reply:
I do not think so, but I might be missing something:
- in `ir_http.py`'s `ModelConverter`'s `to_python`, the `ir.attachment` record is browsed
- I see no override that would verify that the user also has access to the record that is related to the attachment through `res_model` and `res_id`
- the `read` and `exists` methods on `ir.attachment` do not perform such a check either
- `access_ir_attachment_group_user` gives `group_user` full access to `ir.attachment`
- `base_security.xml` does not mention attachment

Reply:
It's done through the `slug()` call in the `to_url()` method, which is trying to read the record.
It's how 403 are shown when trying to access a protected record through (modelconverter) url.
- For instance, you can access the `/shop/furn-8999-canape-trois-places-27` product
- Then create a `Product Template` `ir.rule` like this `[('id', '!=', 27)]` (27 is three seats sofa)
- Now F5 the  `/shop/furn-8999-canape-trois-places-27`  page, it will show an access error without entering the controller. The fact that we are using a modelconverter does the security check for us.

For [tracking](https://discord.com/channels/678381219515465750/687337739129323543/1139134167351820319) purpose, the `ModelConverter` also comes with:
- Automatic multi website support https://github.com/odoo/odoo/commit/e19227d3ba9c9296bfc0c221ac70a863a571b9a6#diff-5e92e473fa4d3da6db7ef727fb217dad51ef6c2383913edca73fe040a23e82c2R737
- Automatic 404 / 403 support, no need to do it every time in controller
- Better / nicer URL to clients, and becomes customizable through the "seo_name" which will replace the name in the slug 
- Without model converter, no sitemap support, it won't be listed unless you define a custom sitemap function
- Without model converter, routes won't be shown in autocomplete (create link etc)

Reply:
I think I am still missing a piece of the puzzle.
I agree that if you call `/web_editor/modify_image/myattachment-123`, it will check that you can access the `ir.attachment` record having `id=123`.
But here the check is on the model record that is referenced from that attachment through its `res_model` and `res_id` fields - and I do not see such a check being automated through the `ModelConverter`.

Reply:
> But here the check is on the model record that is referenced from that attachment 

Indeed, I read the diff too quickly and actually did not realize that part :grimacing: